### PR TITLE
chore(collections): full test coverage

### PR DESCRIPTION
**Pull request by iuioiua**
This change adds some tests and removes some unneeded functionality to bring the test coverage up to 100% for this module.

### CODE REVIEW

File: collections/binary_search_tree.ts [hunk 1]
```diff
@@ -188,11 +188,6 @@ export class BinarySearchTree<T> implements Iterable<T> {
     const replacementDirection: Direction = direction === "left"
       ? "right"
       : "left";
-    if (!node[replacementDirection]) {
```
Comment (kt3k) [inline collections/binary_search_tree.ts:191]: I think this check is still useful for the user who extends this class. You can test this validation by the code like below:

```ts
import { BinarySearchTree  } from "./collections/binary_search_tree.ts";
class MyTree<T> extends BinarySearchTree<T> {
  foo() {
    super.rotateNode(this.root!, "right");
  }
}
const tree = new MyTree();
tree.insert(1);
tree.foo();
```

File: collections/zip.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ import { minOf } from "./min_of.ts";
 export function zip<T extends unknown[]>(
   ...arrays: { [K in keyof T]: T[K][] }
 ): T[] {
-  const minLength = minOf(arrays, (it) => it.length) ?? 0;
+  const minLength = minOf(arrays, (it) => it.length)!;
```
Comment (kt3k) [inline collections/zip.ts:34]: This value can be undefined when the input is an empty array. So this non-null assertion doesn't look good to me. If we want to increase the branch coverage here, we can call `zip()` and that should cover `?? 0` part.

**Review by kt3k — COMMENTED** [review]
Test cases look good to me. Left some comments on the changes in implementation

File: collections/zip.ts (hunk 2 above)
Comment (iuioiua) [reply collections/zip.ts:34]: Ah, my previous assumption was that this wasn't the case. Thanks!

File: collections/binary_search_tree.ts (hunk 1 above)
Comment (iuioiua) [reply collections/binary_search_tree.ts]: I see. Yes, it throws a useful error. I reverted and actually used this snippet as a test instead.

**Review by kt3k — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**